File: EduCollab_benchmark/2_requirement_level_generated_artifacts/python/educollab_flask_req01/models/course_repo.py
from models.db import get_db


def create_course(title, description, created_by):
    db = get_db()
    cursor = db.execute(
        'INSERT INTO courses (title, description, created_by) VALUES (?, ?, ?)',
        (title, description, created_by),
    )
    db.commit()
    return get_course_by_id(cursor.lastrowid)


def list_courses():
    db = get_db()
    return db.execute(
        'SELECT c.course_id, c.title, c.description, c.created_by, c.created_at, u.display_name AS creator_name '
        'FROM courses c JOIN users u ON u.user_id = c.created_by ORDER BY c.course_id DESC'
    ).fetchall()


def get_course_by_id(course_id):
    db = get_db()
    return db.execute(
        'SELECT c.course_id, c.title, c.description, c.created_by, c.created_at, u.display_name AS creator_name '
        'FROM courses c JOIN users u ON u.user_id = c.created_by WHERE c.course_id = ?',
        (course_id,),
    ).fetchone()


def update_course(course_id, title, description):
    db = get_db()
    db.execute(
        'UPDATE courses SET title = ?, description = ? WHERE course_id = ?',
        (title, description, course_id),
    )
    db.commit()
    return get_course_by_id(course_id)


def delete_course(course_id):
    db = get_db()
    db.execute('DELETE FROM courses WHERE course_id = ?', (course_id,))
    db.commit()

File: EduCollab_benchmark/3_file_level_generated_artifacts/javascript/educollab_js_mvc_vul004/src/repos/courseRepo.js
const { db, nowIso } = require('../db');

function createCourse({ title, description, created_by }) {
  const info = db.prepare(
    'INSERT INTO courses (title, description, created_by, created_at) VALUES (?,?,?,?)'
  ).run(title, description, created_by, nowIso());
  return getById(info.lastInsertRowid);
}

function listCourses() {
  return db.prepare(`
    SELECT c.*, u.username as created_by_username
    FROM courses c
    JOIN users u ON u.user_id = c.created_by
    ORDER BY c.course_id DESC
  `).all();
}

function getById(course_id) {
  return db.prepare('SELECT * FROM courses WHERE course_id=?').get(course_id);
}

function updateCourse(course_id, { title, description }) {
  db.prepare('UPDATE courses SET title=?, description=? WHERE course_id=?').run(title, description, course_id);
  return getById(course_id);
}

function deleteCourse(course_id) {
  db.prepare('DELETE FROM courses WHERE course_id=?').run(course_id);
}

module.exports = { createCourse, listCourses, getById, updateCourse, deleteCourse };
